qiskit/primitives/containers/observables_array.py

```diff
@@ -98,7 +98,24 @@ def __repr__(self):
         return prefix + array + suffix
 
     def tolist(self) -> list:
-        """Convert to a nested list"""
+        """Convert to a nested list.
+
+        Similar to Numpy's `tolist` method, the level of nesting
+        depends on the dimension of the observables array. In the
+        case of dimension 0 the method returns a single observable
+        instead of a list.
```

Comment:
Thanks @gadial, in that case, would it incorrect to specify further something like this?

```suggestion
        case of dimension 0 the method returns a single observable 
        (``dict`` in the case of a weighted sum of Paulis) instead of a list.
```